core/src/raw/accessor.rs

```diff
@@ -843,40 +894,101 @@ impl AccessorInfo {
     ///
     /// - name for `s3` => bucket name
     /// - name for `azblob` => container name
-    pub fn name(&self) -> &str {
-        &self.name
+    pub fn name(&self) -> String {
+        match self.inner.read() {
+            Ok(v) => v.name.to_string(),
+            Err(err) => err.get_ref().name.clone(),
+        }
     }
 
     /// Set name of this backend.
-    pub fn set_name(&mut self, name: &str) -> &mut Self {
-        self.name = name.to_string();
+    pub fn set_name(&self, name: &str) -> &Self {
+        match self.inner.write() {
+            Ok(mut v) => v.name = name.to_string(),
+            Err(mut err) => err.get_mut().name = name.to_string(),
+        }
+
         self
     }
 
     /// Get backend's native capabilities.
     pub fn native_capability(&self) -> Capability {
-        self.native_capability
+        match self.inner.read() {
+            Ok(v) => v.native_capability,
+            Err(err) => err.get_ref().native_capability,
+        }
     }
 
     /// Set native capabilities for service.
     ///
     /// # NOTES
     ///
     /// Set native capability will also flush the full capability. The only way to change
-    /// full_capability is via `full_capability_mut`.
-    pub fn set_native_capability(&mut self, capability: Capability) -> &mut Self {
-        self.native_capability = capability;
-        self.full_capability = capability;
+    /// full_capability is via `update_full_capability`.
+    pub fn set_native_capability(&self, capability: Capability) -> &Self {
+        match self.inner.write() {
+            Ok(mut v) => {
+                v.native_capability = capability;
+                v.full_capability = capability;
+            }
+            Err(mut err) => {
+                err.get_mut().native_capability = capability;
+                err.get_mut().full_capability = capability;
+            }
+        }
+
         self
     }
 
     /// Get service's full capabilities.
     pub fn full_capability(&self) -> Capability {
-        self.full_capability
+        match self.inner.read() {
+            Ok(v) => v.full_capability,
+            Err(err) => err.get_ref().full_capability,
+        }
     }
 
     /// Get service's full capabilities.
```

Comment:
please update the docs

Reply:
Thanks for pointing that out. I'll fix it in my next PR.